packages/material-ui/src/Unstable_TrapFocus/Unstable_TrapFocus.js

```diff
@@ -85,6 +87,18 @@ function Unstable_TrapFocus(props) {
       }
 
       if (rootRef.current && !rootRef.current.contains(doc.activeElement)) {
+        // if the focus event is different than the last syntheticEvent from the children, reset
```

Comment:
Having a hard time understanding this branch. Could you explain with JSX and an event order when we reach this branch? Ideally point to an existing test or add a new one for this branch.

Reply:
The condition is trying to capture whether between the time we saved the last focused event target and this invocation of the function, it has changed. We have two scenarios we need to check:
1. the focus event was invoked with different target (that is the first part of the condition)
2. the element that was focused was removed and the interval for the function run out (that is the second part of the condition)

The 2 condition is captured already in the test where an element is removed and the focus moves to body, I will add a test for the other use-case (it may be hard to make sure which exact of these two condition will be caught because the function is invoked in an interval, but let me try to deal with it)

Does this clarifies the use case? I will also try to update the comment to better reflect this.

Reply:
On this note, I would like to test whether just the second part of the condition will be enough, as the interval would anyway invoke this function, but it may have a bit of delay, so it may not be the best idea..

Reply:
The newly added trest together with the previous covers all use casees I believe - focusing an dom outisde the react tree, as well focusing nodes inside portal